Controller/Callback/Offsite.php

```diff
@@ -66,7 +67,7 @@ public function execute()
             return $this->redirect(self::PATH_CART);
         }
 
-        if ($order->getState() !== Order::STATE_PENDING_PAYMENT) {
+        if (! in_array($order->getState(), [ Order::STATE_PENDING_PAYMENT, Order::STATE_PROCESSING ])) {
```

Comment:
this callback offsite is called twice by truemoney.

Reply:
```[2019-06-11 18:25:16] main.DEBUG: offsite callback execute {"state":"pending_payment"} []
[2019-06-11 18:25:18] main.DEBUG: offsite callback execute {"state":"processing"} []```

Reply:
So, the second one comes as a request from Omise servers to the Magento site? Rather than a redirection to the callback URL?

Reply:
☝️

Reply:
@jonrandy why it feels wrong? `Feels wrong processing these two events via this code`
I don't understand where are your worries are coming from.
could you please describe real threats this code brings?
I mean.. what could happen.. and do you have any proofs (ie. omise.co documentation) that that worries could be confirmed.

Reply:
Actually, @jonrandy made a good point.
@jacstn is there any chance that this would conflict with Webhook feature, `charge.complete` event?

Reply:
don't see any problem it could bring to `complete` functionality.

Reply:
webhook is different `controller/action`